src/GitVersionCore/Configuration/ConfigurationProvider.cs

```diff
@@ -31,15 +31,15 @@ public static void ApplyDefaultsTo(Config config)
             var configBranches = config.Branches.ToList();
 
             ApplyBranchDefaults(config, GetOrCreateBranchDefaults(config, "master"), defaultTag: string.Empty, defaultPreventIncrement: true);
-            ApplyBranchDefaults(config, GetOrCreateBranchDefaults(config, "release[/-]"), defaultTag: "beta", defaultPreventIncrement: true);
```

Comment:
Yeah exactly. So I have:

```
branches:
  develop:
    tag: beta
```

The effective config will have both `develop` and `dev(elop)?...`. We need to look for develop, rename it to the new one then delete it